src/java.base/share/classes/jdk/internal/jimage/BasicImageReader.java

```diff
@@ -176,18 +176,9 @@ private void registerIfCracPresent() {
             return;
         }
         try {
-            Object[] priorities = priorityClass.getEnumConstants();
-            if (priorities == null) {
-                return;
-            }
-            Object normalPriority = null;
-            for (int i = 0; i < priorities.length; ++i) {
-                if ("NORMAL".equals(priorities[i].toString())) {
-                    normalPriority = priorities[i];
-                }
-            }
```

Comment:
I'm not sure, maybe this piece of code should be retained as an alternative for compatibility with the current versions? I don't fully understand the scope of compatibility this class has to support (JDK 8, non-CRaC JDKs...)

Reply:
I wouldn't strive for proper CRaC support on any version of JDK, including older CRaCs  - that could explode into code bloat - but if it runs on JDK 8 it should run without errors on older CRaC, too. In other words, if the field is not present, I would silently suppress the `NoSuchFieldException` and return rather than throw `IllegalStateException`.